### DISCUSSION TITLE
2021/03/27 Community Call

### DISCUSSION THREAD

**Discussion by eddiejaoude** [EddieHubCommunity/support#1176] [Community Call]
Sign up for our community call https://www.eddiehub.org/community-call-sign-up

1. Leave a new thread below and we will go through the list in order on our community call (you can read out and discuss your thread on the call, or if you prefer someone else can read it out for you)
2. Feel free to reply to people's threads with thoughts, ideas, support, questions, something you learnt etc. 
3. If you reply in the voice call, please also leave a response on their thread so it is not lost

Please remember our [Code of Conduct](https://github.com/EddieHubCommunity/support/blob/main/CODE_OF_CONDUCT.md).

*Note: this will be locked until the call*

Previous community call discussion thread https://github.com/EddieHubCommunity/support/discussions/994

**Comment by naomi-lgbt** [comment]
Welcome everyone! Hope you all enjoy the call!

Use this comment to introduce yourself - share your github and socials, let us know where you are in your learning journey, and anything else you want us to know!

**Reply by muthuannamalai12** [reply]
Hello Everyone,
I am Muthu Annamalai, currently pursuing BE in Computer Science and Engineering from PEC Chennai. Being a newbie, I am still learning and therefore I try to use every opportunity to know more about technology. Though I am new, I can vouch for the fact that I am a fast learner with a penchant to both learn and unlearn.
Also, you can find me here:
Linkedin Profile: https://www.linkedin.com/in/muthu-annamalai/
Git Hub: https://github.com/muthuannamalai12
Twitter: https://twitter.com/muthuannamalai_

**Reply by Vyvy-vi** [reply]
Hey Everyone-!
I'm Vyvy-vi (Vyom Jain)
Your friendly neighbourhood squinting potato.

My goals, thoughts, and distractions for today's call-
- Learn more about the community's members.
- Find out more long-term project structure preferences - What mix Best Practices and Simplicity?
- Engage in more discussion about our MLH discuss
- Discuss about bot-safety and captcha. What methods are preferred by people and are un-exploitable?
- Ask about Serverless deployment vs Cloud. What's preferred?
- Listen in on Project management tips and discussions

My socials-
Github: https://github.com/vyvy-vi/
Twitter: https://twitter.com/Vyvy_viM
Linkedin: https://www.linkedin.com/in/vyvy-vi/

**Reply by Prathamesh-Shanbhag** [reply]
Hey everyone 👋
This is Prathamesh Shanbhag, I strongly believe in the community and its power. During this journey I wish to connect with many people and help and learn together!
I wouldn't like to align myself to one particular field of technology, because I partake/have interest in multiple kinds. Not only technology, I like working to find solutions to problems, be it real life problems or technical problems and solving them diligently. I am proficient in languages like HTML, CSS, Javascript and I am currently working on my web dev skills!

Currently in the process of learning new tech and constantly upgrade my skills, as well as simultaneously brainstorming about content and trying to put myself out there, much more than I have previously. 
Would love to know all of you!
[Linktree Link to all my Socials!](https://linktr.ee/PrathameshShanbhag)

**Reply by priyadarshu** [reply]
Hi,

This is Priyadarshini from Mumbai, India. I'm currently learning ReactJs, developing an Ecommerce Web application and configuring the github workflows. I am working as a Flutter developer. 

Though I was a part of the community since a long time, I wasn't active much. Finally I started with opensource projects, this week. Thanks Eddie and the community to be very supportive. I really appreciate it. 👏 

- **My first contribution** -> Implementing ``` husky``` to enforce linting and prettier with hooks. 🤩
 
- **Next steps** -> Participate in MLH LHD Share

#### Socials :
LinkedIn -  https://www.linkedin.com/in/priyadarshu/ 
Github - https://github.com/priyadarshu
Twitter - https://twitter.com/priya_darshu_

**Reply by imvisp** [reply]
Hello everyone,
My name is Vishal patadiya and I am second year BCA undergraduate student, I am currently learning python and making a project (A 2D game) and also I am interested in ml and AI love to help others (as much as I know)

- feel free to connect with me

- GitHub - https://github.com/imvisp
- LinkedIn - https://www.linkedin.com/mwlite/in/vishal-patadia-698781200
- Twitter (mostly active) - https://twitter.com/imvishp?s=09

**Reply by felixfaisal** [reply]
Hey Everyone  
My name is Faisal, 3rd year CS undergrad from Presidency University, Bangalore 
I'm currently an MLH Fellow of Spring'21 batch 
I'm an Ex-FOSSE fellow 2020  organised by IIT, Bombay ( Free Open Source Software for Education) 

My Github - https://github.com/felixfaisal
My LinkedIn - https://www.linkedin.com/in/faisal-ahmed-farooq-6395a0174/

**Comment by naomi-lgbt** [comment]
Here's another comment - reply to this one to give us feedback! What you like about our call? What do you think we should improve?

**Comment by naomi-lgbt** [comment]
Let's kick this one off with a question - Talking about GitHub Projects (fancy Kanban, really). 

I use Projects to map out the initial features before I spike out a POC - but I don't see a use-case for longer term development where you might only have a couple of open issues at a time.

Does anyone have examples of how they use GitHub Projects to structure their long term development roadmap?

**Reply by felixfaisal** [reply]
Github projects is great for the following reasons 
- Automated columns 
- Automated movement of todo lists 
- Custom columns 
- Since it's on Github, Most people open Github, so navigating there is easier as opposed to third party kanbans 


Here's an example of how I previously used it with a team of 14 members 
![image](https://user-images.githubusercontent.com/42486737/112730263-9bdf6380-8f56-11eb-8b03-938178618029.png)

**Comment by AayushChandorkar** [comment]
soo i have to ask
i have studied java meaning bluej from class 8 to 12. Now i have started studying Python as a hobby. Which one should I pursue more

**Reply by naomi-lgbt** [reply]
I think that depends on your long-term plan. Are you looking to go in to development professionally? If so, look at what's more "in-demand" for job openings in your area and use that to structure your learning path.

**Reply by AayushChandorkar** [reply]
ok so I have an interest in app development and game development

**Reply by naomi-lgbt** [reply]
I think Java is a bit more popular than Python for app development, but for game development I think the C languages are still the driving force?

**Reply by Vyvy-vi** [reply]
I think, since you're familiar with Java, you should try making a few projects in Java esp. revolving around app development or game development.

For game development, the C languages are the driving force, as Nick said, but Java can be a good choice too.

**Reply by Vyvy-vi** [reply]
You could always switch around. Multi-language discipline is good, and you can switch anytime

I'd suggest look at Java and start implementing projects in that, while making a project or two in Python on the side.

**Reply by AayushChandorkar** [reply]
ohhh okk

**Comment by felixfaisal** [comment]
GitHub actions has become a very useful CI tool lately after it was made available to all 
So, I wanted to know if there are people working on it and face these issues, and knows a workaround xD 
- Need to commit everytime to check the working of yaml file 
- Need to commit changes to fix builds and not possible directly  
- We can't chose to run only  certain workflows for some tests 
There are other CI tools and would love to know how they're better in terms of these

**Reply by naomi-lgbt** [reply]
You shouldn't be committing changes to the `main` branch, IMHO.

For testing new action configuration files, I have a specific repository called `actions-testing` which I can use to play with the config and verify that the YML is correct and the action does what I want.

For restricting workflow runs, if you don't *need* it to run on every PR you can use a different event to trigger the workflow. Or if you want tests only to run when certain files are edited, you can use the `path` config.

**Reply by Vyvy-vi** [reply]
You could make a second testing branch and run the workflow specifically on that branch after edits

**Reply by starlightknown** [reply]
I had this doubt too like how do you test the yml working. This answer helped. I tried on on of my project repos I did it on main but then the build kept failing

**Reply by anushkrishnav** [reply]
My repo to Test https://github.com/anushkrishnav/Tester
My Python Action : https://github.com/TesseractCoding/Issue_Watcher
If you are making an action the you can have a decent working version on main and have a branch called dev to keep pushing your changes.
You can have 2 release (1 for main and 1 for dev ) so that repo depended on your main doesn't get affected you can then test the dev branch by setting  
- uses: {owner name}/{action name} @ { dev branch }
to keep testing your latest changes

**Reply by eddiejaoude** [reply]
I think there are 2 solutions

- use a feature branch with the trigger for that branch, once working, then update the trigger and squash the commits when merging
- local testing (I have not tried it yet)
- you could also add the manual trigger during testing

**Reply by starlightknown** [reply]
Thank you, these answers help a lot

**Comment by Rahma-AlWadhahi** [comment]
I am not sure how to formulate this as a question but I was wondering how would we encourage collaboration and communication if there was some kind of a language barrier.

**Reply by naomi-lgbt** [reply]
I find it helps to use more "simple" language (for lack of a better term). Explain things as clearly and plainly as possible, and avoid too many technical terms.

Also, having patience and understanding is important - I've found that people who struggle with a language are often more shy or afraid of speaking, and I like to encourage people to share their ideas without putting them on the spot.

**Reply by Vyvy-vi** [reply]
~~I think something that is foundational helps.~~
~~Tools like block-based programming would be slightly more intuitive before the use of a language.~~

~~https://scratch.mit.edu/ is a great resource.~~
~~Logic diagram programs like raptor should also be great.~~

I misunderstood 😓 

I think a more visual explanation to convey ideas, should help

**Reply by naomi-lgbt** [reply]
World languages not programming languages :)

**Reply by anushkrishnav** [reply]
My community uses a lot of google translate to help each other

**Reply by eddiejaoude** [reply]
Great question 👍 - the world has got smaller and we are collaborating with many more people from all over the world 💪 

My thoughts

- keep things concise (short sentences)
- use screenshare to help explain

I will do some more research and ask around, because I think this is a great question and I do not have a good answer (yet) 🤓

**Comment by oreo-gif** [comment]
I have participated in GS summer of code, ain't getting a beginner friendly issue and I'm getting confused between so many projects. How to find a good issue ?  :(

**Reply by carrycooldude** [reply]
Approach the maintainer or mentor

**Reply by oreo-gif** [reply]
should I ask on discord?

**Reply by naomi-lgbt** [reply]
I would also encourage you to look for issues labelled `good first issue` or `first timers only`, as these are identified by the maintainers as issues that have a low barrier of entry.

We also have the #good-first-issues channel in our Discord where we share issues.

**Reply by Vyvy-vi** [reply]
I think in cases like that, it would be best to approach the Project Maintainer and have a chat with them. Moreover, asking people in the community, about what project does what, should be helpful

**Reply by carrycooldude** [reply]
> should I ask on discord?

Yup

**Reply by starlightknown** [reply]
If you're talking about girlscript summer of code then they have labeled their issues as level 0,1,2,3. Level 0 and level 1 are beginner friendly. Better to ask the maintainers on discord too.

**Reply by oreo-gif** [reply]
got it. thank you everyone

**Reply by anushkrishnav** [reply]
We are a part of GSSoC'21 https://github.com/TesseractCoding/NeoAlgo
I  maintain this repo. You can contact me for guidance.
Tip: There is a channel  for every repo tag me on NeoAlgo channel we usually shy away from DM but when you post in a public domain we reply for sure

**Reply by felixfaisal** [reply]
So I think one thing that i know in general about contributing to open source first issues, 
Sometimes the tech stack or the library might be unfamiliar, what seems good first issue to the maintainer might not necessarily be an easy fix for you 
So take 3-4 days for testing the application and exploring the bug 
Learn the library to understand only that part of the application 
And take some time to get to know it 
It might take you more than 8-12 days and that's fine but your ultimate contribution will be valuable and you'll better know the architecture 
Also contact mentors regarding any issue you run into

**Comment by carrycooldude** [comment]
What your thoughts on WebAssembly in Machine Learning Or in General Why WebAssembly is there for the web?

**Reply by naomi-lgbt** [reply]
I really haven't explored WASM much, so I can't comment on this beyond the "It seems like the next Big Thing". I'm not sure how far it'll go.

**Reply by felixfaisal** [reply]
This is a really good discussion
Personally I haven't made an application using Web Assembly but I've researched some articles about it I guess which helped me 
- It improves performance 
- It's not as fast as native but it's the closest you can get near native experience on web ( which is huge) 
- So it's really helpful for complex web applications where performance is important but maybe not great for normal website
- If you do enough tweaks, you can get 20x performance compared to previous versions of your web application [Article](https://www.smashingmagazine.com/2019/04/webassembly-speed-web-app/)

Examples
I don't have the links 
- Auto Cad on web 
- Windows 95 on web

**Comment by muthuannamalai12** [comment]
I don’t want to speak this time, but I want to ask a question I have started with web development in recent days to get my hands dirty on it. Can I get some tips on how to master that skill??. I want to be a full-stack web developer.

**Reply by carrycooldude** [reply]
If you want to start my Tips will be works on the MERN stack and make projects on that
- Works on the APIs for fetching the data ,etc...
- Frameworks like Angular, React
- Or Something like REST API or GraphQL

**Reply by naomi-lgbt** [reply]
A great option for a structured learning path is freeCodeCamp, by the way. 😁

**Reply by carrycooldude** [reply]
> A great option for a structured learning path is freeCodeCamp, by the way.

True Nick

**Reply by muthuannamalai12** [reply]
I have just started with frontend

**Reply by oreo-gif** [reply]
agreed @nhcarrigan Freecodecamp is best for that

**Reply by ghost** [reply]
Will recommend to make consistency while coming to your learning part, but talking about mastering the stuffs you are learning, I would recommend to research on what you learn :)
and for getting hands dirty start contributing to open source projects which includes the respective techs you want 👍🏼

**Reply by ghost** [reply]
yeah right @nhcarrigan actually Even I am following that, but would also recommend to do some research on what you learn :)

**Reply by Vyvy-vi** [reply]
I would suggest, give yourself time
It's very interesting to go ahead with the X popular framework, but giving yourself time to establish basics in important stuff like HTML-css, responsive design, Vanilla Javascript would go a long way-!
(I'm saying this, since many people tend to say stuff like- HTML is not a real programming language or something like that, which might mis-guide you into not going through it properly)

Another thing, that I'll like to add is- Make projects, and deploy them. 
(Sharing them with the community, would help you improve and learn)

**Comment by priyadarshu** [comment]
I haven't worked with Digital Ocean or any other Cloud platform to host the web application, how do they differ?
I have only run the applications locally, I would like to get some guidance on it. I have to finalize the Cloud platform, before meeting the Client next week

**Reply by naomi-lgbt** [reply]
It depends on the platform.

DigitalOcean offers Droplets. Droplets are VPS which you manage yourself - you handle the builds, the deploys, etc. Azure is similar.

Heroku, on the other side, will handle deployments and build pipelines for you, which reduces some of the overhead but also gives you less granular control.

**Reply by Vyvy-vi** [reply]
Which stack do you work with?

If you're familiar with running your application in docker, deploying in Kubernetes on DO can be a similar to running local, and would need some yml configs.

**Reply by priyadarshu** [reply]
I'm working on PERN stack

**Reply by ghost** [reply]
Right @nhcarrigan , Let me know if you want to start with Digital Ocean
I can help you out 👍🏼 @priyadarshu

**Reply by priyadarshu** [reply]
@nhcarrigan @Vyvy-vi  What are your thoughts on AWS?

**Reply by naomi-lgbt** [reply]
I haven't explored it myself, but it's expensive. I've heard some horror stories about large bills when someone forgot to flip off a switch.

**Reply by priyadarshu** [reply]
That's scary @nhcarrigan . Would you mind sharing some resources related to Digital Ocean, Heroku and Kubernetes @nhcarrigan @Vyvy-vi @PulkitSinghDev

**Reply by felixfaisal** [reply]
I've used DigitalOcean to host my discord bot and i used azure before, As it was free and pretty useful 
So I guess there's a difference between Digital Ocean and cloud platforms like Azure 
- Digital Ocean is also considered more as a VPS( I don't know why), You just get a shell and you just put your application there but you have to chose your droplet, I chose Docker( so that i can get everything ) 
- Azure seems a bit complicated with many features that are useful and is very strong, but it does involve a learning curve for you to understand all the options that are available, And they have test on the applications itself, Like web application not use any port can lead to the application failing

**Reply by Vyvy-vi** [reply]
AWS S3 buckets are free for 1 year and AWS Lambda(serverless) is great(and somewhat affrodable), but I haven't used AWS, and as Nick said, it's expensive in most cases. 
I personally use Heroku for my deployments. I've tried Cloudfare Workers too and that's great(but that's serverless)
I've heard a lot about DO(and the platform and community is great)

**Reply by naomi-lgbt** [reply]
https://www.nhcarrigan.com/code-snippets/#/notes/do-droplet-setup

Here's the instructions I use when I set up a new DigitalOcean Droplet

**Reply by priyadarshu** [reply]
@felixfaisal Thanks for pointing out the difference

Thanks @Vyvy-vi @nhcarrigan @PulkitSinghDev !!

**Reply by ghost** [reply]
@priyadarshu would like to share this docs of digital ocean with you:-
https://www.digitalocean.com/docs/

**Reply by eddiejaoude** [reply]
I have used so many, but after all these years, I have come back to Digital Ocean.

Many of my big clients want to use AWS because that is what their competitors use. But to be honest for most projects, they are all good solutions.

I now have moved most of my side projects to Digital Ocean and new projects will be on Digital Ocean too

**Reply by priyadarshu** [reply]
That's really great @eddiejaoude !
Thanks for letting me know, I have noted all the points. Will try to convince my Client to use Digital Ocean  :smile:

**Comment by ghost** [comment]
Which language is mostly preferred while getting started with TensorFlow
Python or Javascript ??
do let me know if any reasons behind that 😄

**Reply by naomi-lgbt** [reply]
Pretty sure Python is highly preferred here (TBH I didn't even know JS would *do* TensorFlow).

**Reply by starlightknown** [reply]
There is tensorflow js but I think it's not much used compared to python. I have used tensorflow-js in one of my projects. It was not a big one but yeah it's pretty simple to use

**Reply by ghost** [reply]
ohk thank you @nhcarrigan 
![image](https://user-images.githubusercontent.com/71369943/112730996-a8fe5180-8f5a-11eb-801d-fbecf4412330.png)
😅

**Reply by Vyvy-vi** [reply]
Python is pretty strong when it comes to AI/ML/DS and hence is best for Tensorflow
Moreover, I think there are some wrappers(?) for Tensorflow in python, which make working with it more easier and powerful(pytorch, keras). Moreover, these are popular in python community, so it would be easier to get help

**Reply by Vyvy-vi** [reply]
Moreover, I think Python would lead to higher abstraction, which might help when working in areas, where you don't worry about language issues and get started right away(?)

**Reply by ghost** [reply]
Thank you everyone for sharing your thoughts :smile:

**Reply by felixfaisal** [reply]
Python is strong because of the amount of AI/ML libraries supported by the community as compared to JavaScript. 
JavaScript is great for Frontend work and now even backend, But for complex computations, I suggest Python.

**Comment by ghost** [comment]
Want some insights about LHD and how to have a successful guild because I became a LHD Guild lead for the first time and even joining LHD for the first time 😅

**Reply by starlightknown** [reply]
Help people with their first steps and their next steps since there might be many beginners. Make them feel included. Learning and collaborating together, that'll encourage the team to work together and help each other.

**Reply by ghost** [reply]
Thanks, will try my best

**Comment by eddiejaoude** [comment]
Great questions everyone. I will leave this discussion unlocked so open questions can get more responses - please do not add more questions to this discussion

**Reply by naomi-lgbt** [reply]
If you have additional questions, feel free to ask in our Discord community!
